File: web/src/api/landlord.ts
import apiClient from './request';
import type { Landlord } from '../types/api';

export interface LandlordsQueryResult {
  items: Landlord[];
  total: number;
  page: number;
  limit: number;
}

export interface LandlordQueryParams {
  name?: string;
  phone?: string;
  offset?: number;
  limit?: number;
}

export const landlordApi = {
  list: async (params?: LandlordQueryParams) => {
    const response = await apiClient.get<LandlordsQueryResult>('landlords', { params });
    return response.data;
  },

  get: async (id: string) => {
    const response = await apiClient.get<Landlord>(`landlords/${id}`);
    return response.data;
  },

  create: async (data: { name: string; phone: string; note: string }) => {
    const response = await apiClient.post<Landlord>('landlords', data);
    return response.data;
  },

  update: async (id: string, data: { name: string; phone: string; note: string }) => {
    const response = await apiClient.put<Landlord>(`landlords/${id}`, data);
    return response.data;
  },

  delete: async (id: string) => {
    await apiClient.delete(`landlords/${id}`);
  },
};

File: internal/domain/model/landlord.go
package model

import (
	"time"
)

// Landlord 房东领域模型
type Landlord struct {
	ID        string    `json:"id"`
	Name      string    `json:"name"`
	Phone     string    `json:"phone"`
	Note      string    `json:"note"`
	CreatedAt time.Time `json:"createdAt"`
	UpdatedAt time.Time `json:"updatedAt"`
}

// NewLandlord 创建新房东
func NewLandlord(id, name, phone, note string) *Landlord {
	now := time.Now()
	return &Landlord{
		ID:        id,
		Name:      name,
		Phone:     phone,
		Note:      note,
		CreatedAt: now,
		UpdatedAt: now,
	}
}

// Update 更新房东信息
func (l *Landlord) Update(name, phone, note string) {
	l.Name = name
	l.Phone = phone
	l.Note = note
	l.UpdatedAt = time.Now()
}
